### PR TITLE
[CI] Pin homebrew LLVM@9

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by pprudhvi** [comment]
@ursabot crossbow submit -g gandiva

**Comment by pprudhvi** [comment]
@kszucs, @naman1996 tried to build gandiva with this change and it is failing when it tries to install llvm 10 as part of make. (see cpp/CmakeLists.txt)

p.s., is there a way to trigger gandiva nightly builds using ursabot? we were able to do @ursabot crossbow submit earlier.

**Comment by naman1996** [comment]
@github-actions crossbow submit gandiva*

File: cpp/Brewfile [hunk 1]
```diff
@@ -25,8 +25,7 @@ brew "flatbuffers"
 brew "git"
 brew "glog"
 brew "grpc"
-brew "llvm"
-brew "llvm@8"
```
Comment (kou) [inline cpp/Brewfile:29]: We need LLVM 8 for Clang Tools such as `clang-tidy`.

**Comment by kou** [comment]
@github-actions crossbow submit -g gandiva

**Comment by pprudhvi** [comment]
@github-actions crossbow submit -g gandiva

File: cpp/Brewfile (hunk 1 above)
Comment (kszucs) [reply cpp/Brewfile:29]: @kou how about separating the Brewfile required for CI and for local development?

Comment (kou) [reply cpp/Brewfile:29]: It's OK to me.
But I'm not sure we can maintain both of them.

**Comment by pprudhvi** [comment]
@kszucs , @kou Can we make gandiva-jar-osx build pass before merging this? If llvm10 is not available, cmake should pickup llvm9, somehow this is not happening.

**Comment by kou** [comment]
We don't need this once #8521 is merged.

@pprudhvi BTW, could you fix the gandiva-jar-xenial failure?
It seems that this is caused by https://github.com/apache/arrow/commit/f1f4001660a634c311a1580160823bc645806273 .

**Comment by pprudhvi** [comment]
@kou we are working on it

**Comment by kou** [comment]
Great!

FYI: We can't use LLVM 11 on Ubuntu Xenial because g++ 5 on Ubuntu Xenial is old to use LLVM 11. LLVM 11 requires C++ 14 but g++ 5's C++ 14 support is limited.
We're using LLVM 8 for the deb package for Ubuntu Xenial: https://github.com/apache/arrow/blob/master/dev/tasks/linux-packages/apache-arrow/apt/ubuntu-xenial/Dockerfile#L32-L35

**Comment by kou** [comment]
I close this because #8521 is merged.

**Pull request closed without merging**